airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx

```diff
@@ -0,0 +1,105 @@
+import { load, YAMLException } from "js-yaml";
+import React, { useContext, useEffect, useState } from "react";
+import { useDebounce, useLocalStorage } from "react-use";
+
+import {
+  StreamReadRequestBodyConfig,
+  StreamsListReadStreamsItem,
+  StreamsListRequestBodyManifest,
+} from "core/request/ConnectorBuilderClient";
+
+import { useListStreams } from "./ConnectorBuilderApiService";
+import { template } from "./YamlTemplate";
+
+interface Context {
+  yamlManifest: string;
+  jsonManifest: StreamsListRequestBodyManifest;
+  streams: StreamsListReadStreamsItem[];
+  selectedStream: StreamsListReadStreamsItem;
+  configString: string;
+  configJson: StreamReadRequestBodyConfig;
+  setYamlManifest: (yamlValue: string) => void;
+  setSelectedStream: (streamName: string) => void;
+  setConfigString: (configString: string) => void;
+}
+
+export const ConnectorBuilderStateContext = React.createContext<Context | null>(null);
```

Comment:
I created this context to hold all of the connector builder state that needs to be set and accessed in multiple places around the application. Currently, that is just the connector manifest value, the testing config JSON blob, and the selected stream. 

I expect that we may need to split this context out into multiple different ones at some point as we add more complexity to the app, but I decided to keep it in one context for now since it is still fairly simple.